### PR TITLE
Adds support for pcg32 library

### PR DESCRIPTION
**Pull request by zheng95z**

### CODE REVIEW

File: packages/p/pcg32/xmake.lua [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package("pcg32")
+    set_homepage("https://github.com/wjakob/pcg32")
+    set_description("Tiny self-contained C++ version of the PCG32 pseudorandom number generator")
+    set_license("Apache-2.0")
+
+    add_urls("https://github.com/wjakob/pcg32.git")
```
Comment (waruqi) [inline packages/p/pcg32/xmake.lua:6]: please add a fakse version if no any releases/tags

```lua
add_versions("2016.06.07", "70099eadb86d3999c38cf69d2c55f8adc1f7fe34")
```

**Comment by zheng95z** [comment]
I also changed the way of adding a head-only library since a lot of people prefer `pcg32/pcg32.h` instead of `pcg32.h`.

File: packages/p/pcg32/xmake.lua (hunk 1 above)
Comment (zheng95z) [reply packages/p/pcg32/xmake.lua:6]: Done.

**Comment by waruqi** [comment]
> I also changed the way of adding a head-only library since a lot of people prefer `pcg32/pcg32.h` instead of `pcg32.h`.
> 
> And I'm wondering whether there is a way can do that inside:
> 
> ```lua
> io.writefile("xmake.lua", [[
>     target("pcg32")
>         set_kind("headeronly")
>         add_headerfiles("pcg32.h")
> ]])
> import("package.tools.xmake").install(package)
> ```

need only os.cp

**Pull request merged**